cookbook/import_export/clean-csv-file-during-product-import.rst

```diff
@@ -0,0 +1,112 @@
+How to Clean a CSV File During a Products Import
+================================================
+
+The foundations of connector creation have been covered in the previous chapter (cf :doc:`/cookbook/import_export/create-connector`). With the following hands-on practice, we will create our own specific connector.
+
+To stay focused on the main concepts, we will implement the simplest connector possible by avoiding to use too many existing elements.
+
+Our use case is to clean the following CSV file when importing products:
+
+.. literalinclude:: ../../src/Acme/Bundle/CsvCleanerConnectorBundle/Resources/fixtures/products.csv
+   :language: xml
+   :linenos:
+
+Here, we want remove the prefix ``uselesspart-`` in the sku before to run a classic import.
+
+We assume that we're using a standard edition with the ``icecat_demo_dev`` data set, ``sku`` and ``name`` already exist as real attributes of the PIM.
+
+.. note::
+
+    The code inside this cookbook entry is available in the src directory, you can clone pim-docs (https://github.com/akeneo/pim-docs) and use a symlink to make the Acme bundle available in the `src/`.
+
+Create our Connector
+--------------------
+
+Create a new bundle:
+
+.. literalinclude:: ../../src/Acme/Bundle/CsvCleanerConnectorBundle/AcmeCsvCleanerConnectorBundle.php
+   :language: php
+   :linenos:
+
+Register the bundle in AppKernel:
+
+.. code-block:: php
+    :linenos:
+
+    public function registerBundles()
+    {
+        // ...
+            new Acme\Bundle\CsvCleanerConnectorBundle\AcmeCsvCleanerConnectorBundle(),
+        // ...
+    }
+
+Configure our Job
+-----------------
+
+Configure a job in ``Resources/config/batch_jobs.yml``:
+
+.. literalinclude:: ../../src/Acme/Bundle/CsvCleanerConnectorBundle/Resources/config/batch_jobs.yml
+   :language: yaml
+   :linenos:
+   :lines: 1-13
+
+Here we create an import job which contains a single step: `import`.
+
+The default step is ``Akeneo\Bundle\BatchBundle\Step\ItemStep``.
+
+An item step is configured with 3 elements, a reader, a processor and a writer.
+
+Here, we'll use a custom processor service, named ``acme_csvcleanerconnector.processor.denormalization.product.flat`` but we'll continue to use default reader and writer.
```

Comment:
service named ``acme_csvcleanerconnector.processor.denormalization.product.flat``,(comma here) but we